src/services/authServices.ts

```diff
@@ -0,0 +1,122 @@
+import brcypt from 'bcrypt';
+
+import Users from '../types/users';
+import UserModal from '../Models/user';
+
+const HASH_SALT_ROUND = 10;
+
+async function getUserByLoginCred(email: string) {
+  const result = await UserModal.find({ email: email }).populate('boards', 'title').select('-__v');
+  return result;
+}
+
+async function authenticateUser(password: string, hashedPassword: string): Promise<boolean> {
+  try {
+    const result = await brcypt.compare(password, hashedPassword);
```

Comment:
You can also separate bcrypt password hashing logic in separate files e.g. `hash.ts` inside `utils`.

- It separates implementation.
- Another reason for wrapping bcrypt inside `hash.ts` is there might be a case you might need to change or upgrade the package at some point in time. In that case, you can simply make changes in a single file instead of finding all the implementation.